apps/web/src/pages/settings/tabs/ApiKeysCard.tsx

```diff
@@ -1,23 +1,37 @@
-import React from 'react';
+import { useState, useEffect } from 'react';
 import { ActionIcon, InputWrapper } from '@mantine/core';
 import { useClipboard } from '@mantine/hooks';
-import { useQuery } from 'react-query';
+import { useQuery, useMutation } from 'react-query';
 import { Input, Tooltip } from '../../../design-system';
 import { Check, Copy } from '../../../design-system/icons';
 import { getApiKeys } from '../../../api/environment';
 import styled from 'styled-components';
 import { inputStyles } from '../../../design-system/config/inputs.styles';
 import { useEnvController } from '../../../store/use-env-controller';
+import { Regenerate } from './components/Regenerate';
 
 export const ApiKeysCard = () => {
+  const [apiKeys, setApiKeys] = useState<{ key: string }[]>([]);
   const clipboardApiKey = useClipboard({ timeout: 1000 });
   const clipboardEnvironmentIdentifier = useClipboard({ timeout: 1000 });
-  const { data: apiKeys } = useQuery<{ key: string }[]>('getApiKeys', getApiKeys);
+  const { mutateAsync: getApiKeysMutation } = useMutation<
+    { key: string }[],
+    { error: string; message: string; statusCode: number }
+  >(getApiKeys);
+  const { data } = useQuery<{ key: string }[]>('getApiKeys', getApiKeys);
+  useEffect(() => {
+    setApiKeys(data || []);
+  }, [data]);
   const { environment } = useEnvController();
 
-  const apiKey = apiKeys?.length ? apiKeys[0].key : '';
+  const apiKey = apiKeys.length ? apiKeys[0].key : '';
   const environmentIdentifier = environment?.identifier ? environment.identifier : '';
 
+  async function fetchApiKeys() {
+    const keys = await getApiKeysMutation();
+    setApiKeys(keys);
+  }
```

Comment:
@florian-lefebvre this could be achieved also calling refetch with react-query instead of wrapping a mutation around it. I will submit a PR in a sec so you can see how it is also possible to do.

Reply:
Great thanks, I'm not familiar with react query

Reply:
Added a commit here: https://github.com/novuhq/novu/pull/1292/commits/41eb915f75b19395452153c84dd56adf7e472f64

Reply:
I saw it yes, great!